packages/core/rsc/streamable.tsx

```diff
@@ -377,31 +356,55 @@ function createStreamableValueImpl<T = any, E = any>(initialValue?: T) {
     append(value: T) {
       assertStream('.append()');
 
-      if (
-        typeof currentValue !== 'string' &&
-        typeof currentValue !== 'undefined'
-      ) {
-        throw new Error(
-          `.append(): The current value is not a string. Received: ${typeof currentValue}`,
-        );
-      }
-      if (typeof value !== 'string') {
+      if (typeof currentValue === 'undefined') {
+        if (typeof value === 'string' || isValidElement(value)) {
+          currentPatchValue = undefined;
+          currentValue = value;
+        } else {
+          throw new Error(
```

Comment:
I have introduced dedicated errors so we can have error pages for troubleshooting, e.g. https://sdk.vercel.ai/docs/troubleshooting/ai-sdk-errors/ai-invalid-data-content-error

do you think it's worth adding dedicated errors here?

Reply:
Yes! We should convert all the common errors to dedicated error codes, with clear example and explanation.